src/typeguard/_functions.py

```diff
@@ -36,16 +41,41 @@
 
 
 @overload
-def check_type(value: object, expected_type: type[T]) -> T:
+def check_type(
+    value: object,
+    expected_type: type[T],
+    *,
+    forward_ref_policy: ForwardRefPolicy = ...,
+    typecheck_fail_callback: TypeCheckFailCallback | None = ...,
+    collection_check_strategy: CollectionCheckStrategy = ...,
+) -> T:
     ...
 
 
 @overload
-def check_type(value: object, expected_type: Any) -> Any:
+def check_type(
+    value: object,
+    expected_type: Any,
+    *,
+    forward_ref_policy: ForwardRefPolicy = ...,
+    typecheck_fail_callback: TypeCheckFailCallback | None = ...,
+    collection_check_strategy: CollectionCheckStrategy = ...,
+) -> Any:
     ...
 
 
-def check_type(value: object, expected_type: Any) -> Any:
+def check_type(
+    value: object,
+    expected_type: Any,
+    *,
+    forward_ref_policy: ForwardRefPolicy = TypeCheckConfiguration.forward_ref_policy,
```

Comment:
Should probably create a default instance of `TypeCheckConfiguration` instead of referencing class attributes, because this would most likely break if a definition is using `dataclasses.field()`.

Reply:
Maybe `forward_ref_policy: ForwardRefPolicy = TypeCheckConfiguration().forward_ref_policy` ?

Reply:
Linters complained about instantiations in defaults, so I did almost as suggested but broke out to a module-local variable: `_default_config`.

Reply:
What was the complaint?

Reply:
For this diff:

```diff
diff --git a/src/typeguard/_functions.py b/src/typeguard/_functions.py
index db51c1e..66cc92e 100644
--- a/src/typeguard/_functions.py
+++ b/src/typeguard/_functions.py
@@ -73,7 +73,7 @@ def check_type(
     *,
     forward_ref_policy: ForwardRefPolicy = _default_config.forward_ref_policy,
     typecheck_fail_callback: (TypeCheckFailCallback | None) = (
-        _default_config.typecheck_fail_callback
+        TypeCheckConfiguration().typecheck_fail_callback
     ),
     collection_check_strategy: CollectionCheckStrategy = (
         _default_config.collection_check_strategy
```

I get this:

```sh
$ pre-commit run ruff --all-files
ruff.....................................................................Failed
- hook id: ruff
- exit code: 1

src/typeguard/_functions.py:76:9: B008 Do not perform function call `TypeCheckConfiguration` in argument defaults
Found 1 error.
```

Reply:
That seems to come from a flake8-bugbear rule. It's safe to ignore. You can change add `"B008"` to `tool.ruff.ignore` in `pyproject.toml` where it already has `"PGH001"`.